mdtk/tests/test_midi.py

```diff
@@ -0,0 +1,199 @@
+import pandas as pd
+import os
+import pretty_midi
+import shutil
+
+import mdtk.midi as midi
+
+USER_HOME = os.path.expanduser('~')
+TEST_CACHE_PATH = os.path.join(USER_HOME, '.mdtk_test_cache')
+MIDI_PATH = f"mdtk{os.path.sep}tests{os.path.sep}test.mid"
+
+def test_df_to_csv():
+    notes = []
+    notes.append({'onset': 0.0,
+                  'track': 1,
+                  'pitch': 42,
+                  'dur': 1.5})
+    notes.append({'onset': 0.3,
+                  'track': 1,
+                  'pitch': 40,
+                  'dur': 1.07})
+    notes.append({'onset': 0.3,
+                  'track': 2,
+                  'pitch': 56,
+                  'dur': 1.11})
+    
+    df = pd.DataFrame(notes)
+    
+    csv_name = TEST_CACHE_PATH + os.path.sep + 'test.csv'
+    try:
+        os.remove(csv_name)
+    except:
+        pass
+    
+    midi.df_to_csv(df, csv_name)
+    
+    assert os.path.exists(csv_name), ('No csv created.')
+    
+    # Check that notes were written correctly
+    with open(csv_name, 'r') as file:
+        for i, line in enumerate(file):
+            split = line.split(',')
+            assert float(split[0]) == notes[i]['onset'], ("Onset time of " +
```

Comment:
If you were to make my change above to `notes`, you'd have to change these to `notes['onset'][i]` etc.

Reply:
Gotcha.